test/e2e/antreapolicy_test.go

```diff
@@ -4180,6 +4280,43 @@ func doProbe(t *testing.T, data *TestData, p *CustomProbe, protocol AntreaPolicy
 	}
 }
 
+func doEvaluation(t *testing.T, data *TestData, npEvaluation *NPEvaluation) {
+	log.Infof("Running networkpolicyevaluation tests")
+	antreaController, err := data.getAntreaController()
+	require.NoError(t, err, "Error occurred when trying to get the Antrea Controller")
+	antctlArgs := []string{"antctl", "query", "networkpolicyevaluation"}
+	getRealizedPod := func(label Pod) string {
+		realizedPod, err := k8sUtils.GetPodByLabel(label.Namespace(), label.PodName())
+		require.NoError(t, err, "Failed to get Pod %s", label.String())
+		return realizedPod.Namespace + "/" + realizedPod.Name
+	}
+	processResponse := func(stdout string) map[string]string {
+		// the first row is the header row, the second row contains the actual value
+		rows := strings.Split(strings.TrimSpace(stdout), "\n")
+		require.Len(t, rows, 2)
+		headerFields := strings.Fields(rows[0])
+		valueFields := strings.Fields(rows[1])
+		require.Equal(t, len(headerFields), len(valueFields))
+		antctlData := make(map[string]string)
+		for idx, name := range headerFields {
+			antctlData[name] = valueFields[idx]
+		}
+		return antctlData
+	}
+
+	for _, eval := range npEvaluation.Assertions {
+		cmds := append(antctlArgs, "-S", getRealizedPod(eval.Source), "-D", getRealizedPod(eval.Destination))
+		log.Tracef("Running command: %s", strings.Join(cmds, " "))
+		stdout, stderr, err := data.RunCommandFromPod(antreaNamespace, antreaController.Name, "antrea-controller", cmds)
+		if !assert.NoError(t, err, "Failure -- Command was not successful:\n%s", antctlOutput(stdout, stderr)) {
+			continue
+		}
+		gotEval := processResponse(stdout)
+		assert.Equal(t, eval.NPName, gotEval["NAME"], "Failure -- wrong policy rule matched for evaluation")
+		assert.EqualValues(t, eval.Action, gotEval["ACTION"], "Failure -- wrong policy rule action for evaluation")
```

Comment:
```suggestion
		assert.Equal(t, eval.NPName, gotEval["NAME"], "Policy name mismatch in NetworkPolicyEvaluation response")
		assert.EqualValues(t, eval.Action, gotEval["ACTION"], "Policy rule action mismatch in NetworkPolicyEvaluation response")
```

IMO, "wrong policy rule matched" is not accurate because policy rule != policy. A policy can have multiple rules, and we are checking the policy name here not the policy rule name? I don't think we even include the rule name in the API response, we just include the rule index.

Reply:
Yes, so perhaps I can also check rule index. As for `Failure -- ` prefix, I was trying to keep it consistent with [other error messages](https://github.com/antrea-io/antrea/blob/d4922f06f75a48572d8e8bcb2daf596524ab3f7e/test/e2e/antreapolicy_test.go#L4145) and easier to search if that makes sense.

Reply:
Sure, you can keep the formatting that you want as long as you replace "policy rule" with "policy name" in the first message.

Reply:
I replaced the message, as for rule index, since all of them are `0` or `maxint32` (even for the test-case that checks rule priority), I'll not complicate the tests.